### ISSUE TITLE
How to abbreviate file names in recentf list?

### ISSUE THREAD

**Issue by rathagrimes** [abo-abo/swiper#2431]
Hello, I'm using `counsel-recentf` and unlike ivy's find-file, the list of files is shown with absolute paths. This can be a problem on Android termux, because the home directory is quite long (see attached screenshot) so the actual filenames are sometimes occluded.

I searched through the documentation, issues, and code, and tried setting `(setq ivy-virtual-abbreviate 'abbreviate)` but this didn't do anything, presumably because I am not enabling `(setq ivy-use-virtual-buffers t)` because I don't actually want my find-file and my recentf to be mingled, I like them as separate commands.

Is there a way to ask Ivy to apply `abbreviate-file-name` to the filenames from recentf?

I saw an add-on package [here](https://github.com/Yevgnen/ivy-rich) that offered this setting: `(setq ivy-rich-path-style 'abbrev)` but I haven't tried this package because I am reluctant to add this much complexity to see if this one thing is addressed.

Thanks!

![Screenshot_20200125-221827](https://user-images.githubusercontent.com/10936877/73131575-5033d180-3fc2-11ea-8bb0-0ef4f81893b3.png)

**Comment by rathagrimes** [comment]
I found my own answer so I'll close this and maybe it will help others.

The magic setting is `(add-to-list 'recentf-filename-handlers 'abbreviate-file-name)`. I had abandoned this some time ago because I could never get it to work with the vanilla recentf display buffer, so I had written a hack to transform the data in the recentf buffer in order to get the abbreviation. But the setting works with ivy!
